linter/lints/cpcps/lint_crl_has_idp.go

```diff
@@ -69,91 +88,100 @@ func (l *crlHasIDP) Execute(c *x509.RevocationList) *lint.LintResult {
 		}
 	}
 
-	// Ensure that the DistributionPoint is a reasonable URI. To get to the URI,
+	// We read these booleans as a byte and ensure the value is 0xFF because
+	// cryptobyte.ReadASN1Boolean can't handle the custom encoding rules for the
+	// [1] and [2] tagged fields referenced above. X.690 (07/2002) section 8.2
+	// states that a boolean set to true will have contents FF.
+	// https://www.itu.int/rec/T-REC-X.690-200207-S/en
+	onlyContainsUserCerts := make([]byte, 0)
+	onlyContainsCACerts := make([]byte, 0)
+
+	// Ensure that the distributionPoint is a reasonable URI. To get to the URI,
 	// we have to step inside the DistributionPointName, then step inside that's
 	// FullName, and finally read the singular SEQUENCE OF GeneralName element.
```

Comment:
Move this comment inside this outermost `if`. It doesn't apply to the logic of "are we looking at a Subscriber CRL or an Issuer CRL", which is currently missing a comment at all. I'd move this existing comment to between lines 102 and 103 (i.e. the start of the distributionPoint-parsing logic), and replace it with something like:
> If either the distributionPoint field or the onlyContainsUserCerts field is present, assume we're dealing with a CRL containing Subscriber Certs.